src/main/scala/org/ergoplatform/modifiers/mempool/UnconfirmedTransaction.scala

```diff
@@ -0,0 +1,62 @@
+package org.ergoplatform.modifiers.mempool
+
+import scorex.core.serialization.ScorexSerializer
+import scorex.util.ScorexLogging
+import scorex.util.serialization.{Reader, Writer}
+
+
+case class UnconfirmedTransaction(
+  transaction: ErgoTransaction,
+  lastCost: Option[Int],
+  createdTime: Long,
+  lastCheckedTime: Long,
+  transactionBytes: Option[Array[Byte]])
+//) extends Transaction
+  extends ScorexLogging {
+//  with ContainsModifiers[UnconfirmedTransaction] {
+
+  def updateCost(cost: Int): UnconfirmedTransaction = {
+    copy(lastCost = Some(cost), lastCheckedTime = System.currentTimeMillis())
+  }
+
+//  override type M = UnconfirmedTransaction
+//
+//  override def serializer: ScorexSerializer[UnconfirmedTransaction] = UnconfirmedTransactionSerializer
+//
+//  override val messageToSign: Array[Byte] = _
+}
+
+object UnconfirmedTransaction {
+
+  def apply(tx: ErgoTransaction): UnconfirmedTransaction = {
+    UnconfirmedTransaction(
+      tx,
+      None,
+      0,
+      0,
+      Some(ErgoTransactionSerializer.toBytes(tx))
+    )
+  }
+}
+
+object UnconfirmedTransactionSerializer extends ScorexSerializer[UnconfirmedTransaction] {
```

Comment:
Why do you need for serializer ?

Reply:
Removed as per our discussion.  Added JSON codec instead.